src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -533,7 +517,12 @@ case class IndexLogEntry(
     case _ => false
   }
 
-  def numBuckets: Int = derivedDataset.properties.numBuckets
+  def numBuckets: Int = {
```

Comment:
cc @imback82 @clee704 what do you say about this approach?

Reply:
I thought the class name is shown in JSON so that we could utilize it. Just forget about the suggestion.

Then we could move this function to under HashPartition and add the below function.

```scala
def propAs[T]: T = {
  derivedDataset.properties.asInstanceOf[T]
}
```

```scala
index.propAs[HashPartition].numBuckets
```